### PR TITLE
RI-8317 Detect and highlight large vector embeddings in the query editor

### PR DESCRIPTION
**Pull request by dantovska**
# What

Adds detection for large vector embeddings in the query editor, plus a subtle highlight so they're visible.

- **Pure detector** (`utils/monaco/vectorEmbeddingUtils.ts`): `detectVectorEmbeddings(query)` scans query text and returns a mark per embedding. Two formats:
  - binary FLOAT32 `\x` blobs, validated with the existing `isBinaryVector` heuristic;
  - numeric arrays with ≥ 10 elements.
  Each mark carries range, format, byte size, dimensions, a first-3/last-2 preview, and the FT.SEARCH `PARAMS` argument name when applicable (located via `splitQueryByArgs`). Thresholds are named constants; marks/types live in their own file.
- **`useVectorEmbeddingMarks`** hook exposes the marks, recomputed on query change.
- **`VectorEmbeddingHighlight`** — a self-contained component (own styles + detection + Monaco decorations) rendered next to the editor in both Workbench and Vector Search. It paints a subtle grey background behind each embedding, with an i18n hover showing dimensions and byte size. The shared `CodeEditor` / `useQueryEditor` are left untouched so the embedding logic can grow independently.

# Testing

**Unit tests** — `utils/tests/monaco/vectorEmbeddingUtils.spec.ts` cover both formats, multiple embeddings in one query, threshold edges, non-vector strings, and `PARAMS` name extraction:

```
node node_modules/.bin/jest 'redisinsight/ui/src/utils/tests/monaco/vectorEmbeddingUtils.spec.ts' -c jest.config.cjs
```

**Manual** — open a database → Workbench or Vector Search editor and paste the query below:

```
FT.SEARCH idx:bikes_vss "*=>[KNN 3 @description_embeddings $my_blob AS score ]" RETURN 4 score brand type description PARAMS 2 my_blob "\x98=\xd0\xbb\x94\xeb\xdb<C\x85\xf3:@\xdf$=\xd8\xf5\xbf\xbc\xa2\xdb\x04=3\x10\x22<\x8f\xc0)\xbd" SORTBY score DIALECT 2
```

The `my_blob` embedding value should render with a subtle grey background (slightly greyed out on both light and dark themes), and hovering it should show the dimensions and byte size.

---

Closes #RI-8317

---

> [!NOTE]
> **Low Risk**
> Editor-only UI and parsing for display; no changes to query execution, auth, or data persistence.
> 
> **Overview**
> Adds **detection and in-editor highlighting** for large vector embeddings in Workbench and Vector Search query editors, so long `\x` blobs and numeric arrays are easier to spot in FT/KNN queries.
> 
> A new **`detectVectorEmbeddings`** scanner finds quoted binary FLOAT32 strings (via existing `isBinaryVector`) and numeric arrays with at least 10 elements, and can attach **FT.SEARCH `PARAMS` argument names** when the embedding is passed as a param value. **`VectorEmbeddingHighlight`** sits beside the shared `CodeEditor`, uses Monaco inline decorations (`.monaco-vector-embedding`), and shows an i18n hover with dimensions and byte size.
> 
> The feature is kept out of the core `CodeEditor` / `useQueryEditor` path; editors only pass `monacoObjects` and `query` from context. Unit tests cover detection edge cases and PARAMS name mapping.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d57ca217c7f8577f674b8e0df70ac4a79fa8f8e0. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: redisinsight/ui/src/components/base/code-editor/CodeEditor.styles.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ const monacoStyle = (theme: Theme) => {
     --monaco-color-submit: ${colors.icon.success500};
     --monaco-color-bg: ${bgColor};
     --monaco-color-params: ${colors.text.discovery200};
-    
+
```
Comment (valkirilov) [inline redisinsight/ui/src/components/base/code-editor/CodeEditor.styles.ts:48]: 😀

**Review by pawelangelow — APPROVED** [review]
